oasislmf/api_client/session_manager.py

```diff
@@ -0,0 +1,150 @@
+#!/bin/env python
+
+import requests
+from requests import codes as status 
+from requests import Session
+from requests.exceptions import *
+
+from posixpath import join as urljoin
+import time
+import logging
+
+from ..utils.exceptions import OasisException
+
+class SessionManager(Session):
+    def __init__(self, api_url, username, password, timeout=1, retries=3, retry_delay=1, logger=None, **kwargs):
+        super(SessionManager, self).__init__(**kwargs)
+        self.logger = logger or logging.getLogger()
+
+        # Extended class vars
+        self.tkn_access  = None
+        self.tkn_refresh = None
+        self.url_base    = urljoin(api_url, '')
+        self.timeout     = timeout
+        self.retry_max   = retries 
+        self.retry_delay = retry_delay
+
+        # Base Session class vars 
+        self.headers = {
+            'authorization': '',
+            'accept': 'application/json',
+            'content-type': 'application/json',
+        }
+
+        # Check connectivity & authentication
+        self.health_check()
+        self.__get_access_token(username, password) 
+
+
+    def __get_access_token(self, username, password):
+        url  = urljoin(self.url_base,'refresh_token/')
+        r = self.post(url, json={"username": username, "password": password})
+
+        if r.ok:
+            self.tkn_access  = r.json()['access_token']
+            self.tkn_refresh = r.json()['refresh_token']
+            self.headers['authorization'] = 'Bearer {}'.format(self.tkn_access)
+        else:
+            err_msg = 'Login failed: {}'.format(r.json())
+            #self.logger.info(err_msg)
+            raise OasisException(err_msg)
+        return r
+
+    def _refresh_token(self):
+        self.headers['authorization'] = 'Bearer {}'.format(self.tkn_refresh)
+        url = urljoin(self.url_base,'access_token/')
+        r = super(SessionManager, self).post(url, timeout=self.timeout)
+        if r.status_code == status.ok:
+            self.tkn_access  = r.json()['access_token']
+            self.headers['authorization'] = 'Bearer {}'.format(self.tkn_access)
+        else:
+            err_msg = 'Token refresh error: {}'.format(r.text)
+            #self.logger.info(err_msg)
+            raise OasisException(err_msg)
+        return r
+
+
+
+    ## Connection Error Handler     
+    def __recoverable(self, error, url, request, counter=1):
+        if not (counter < self.retry_max):
+            self.logger.info("Max retries of '{}' reached".format(self.retry_max))
+            return False
+
+        if hasattr(error,'status_code'):
+            self.logger.debug("Connection Error Handler: {}".format(error.status_code))
+            if error.status_code in [502, 503, 504]:
+                error = "HTTP %s" % error.status_code
```

Comment:
I note you're using the old string formatting operator (`%`), but it may be best to use the new one `{}`. So here this would be

    error = "HTTP {}".format(error.status_code)

Same for all other places where you're using `%`.